src/Lecture1.hs

```diff
@@ -119,4 +147,16 @@ and lower than 6 elements (4, 5, 6, 7, 8 and 9).
 
 🕯 HINT: Use recursion to implement this function.
 -}
-lowerAndGreater n list = error "TODO"
+lowerAndGreater :: Int -> [Int] -> [Char]
+lowerAndGreater n list = 
+    let calculateMin x acc l = if null l then acc else calculateMin x (if (head l) < x then (acc + 1) else acc) (tail l)
+        calculateMax x acc l = if null l then acc else calculateMax x (if (head l) > x then (acc + 1) else acc) (tail l)
```

Comment:
The implementation is correct. But it traverses the list twice:

1. First time to find the number of elements smaller.
2. Second time to find bigger elements.

Generally, it's preferred to traverse lists only once. Such an approach leads to the automatic streaming of lists. Could you change your implementation to traverse the list only once and find both smaller and bigger numbers within a single list traversal?

Reply:
The most straidforward way to avoid this for me was to write a traversal which returns min and max. But at first I had thought that we didn' yet learn how to do this in Haskell (to implement a type for that). 
Finally I was able to implement it with a tuple:)

Thank you so much for this clarification! I also updated version with 'where' clause to avoid code repetition and had much fun with passing the operator through the recursive function :)

Reply:
> Congrats! The first homework is complete 👏🏼 Hope you enjoyed the course so far. Keep up the great work, and feel free to ask any questions if you have 👌🏼
> 
> I left some feedback. Feel free to improve the solutions if you want, or merge as it is 🙂

Thank you SO MUCH for those comments!

I pushed the updated version, it is mostly fixing your comments, but also it contains upgraded versions for the "lowerAndGreater" function.